### PR TITLE
Branch metrics integration

### LINKED ISSUE #16: Branch metrics analytics feature request
Branch metrics needs additional setup.

Can we have it in the plugin ?

### PR DESCRIPTION
**Pull request by shashikantdurge**
Fixes #16

### CODE REVIEW

File: ios/flutter_segment.podspec [hunk 1]
```diff
@@ -16,7 +16,7 @@ A new flutter plugin project.
   s.public_header_files = 'Classes/**/*.h'
   s.dependency 'Flutter'
   s.dependency 'Analytics', '3.8.2'
-
+  s.dependency 'Segment-Branch'
```
Comment (MaiKaY) [inline ios/flutter_segment.podspec:19]: Could you please set a fixed version here.

File: android/build.gradle [hunk 2]
```diff
@@ -34,5 +34,6 @@ android {
 
     dependencies {
         implementation 'com.segment.analytics.android:analytics:4.7.0'
+        implementation 'io.branch.segment.analytics.android.integrations:branch:+'
```
Comment (MaiKaY) [inline android/build.gradle:37]: Could you please set a fixed version here.

**Comment by MaiKaY** [comment]
Hi @shashikantdurge,
thanks a lot for your PR!

Do you think it makes sense to configure this branch.io integration, similar to `TRACK_APPLICATION_LIFECYCLE_EVENTS`?

**Comment by shashikantdurge** [comment]
**ENABLE_BRANCH_METRICS**  is fine ? 
default value - _false_

**Comment by MaiKaY** [comment]
What is about `ENABLE_BRANCH_IO_INTEGRATION`?

File: android/build.gradle (hunk 2 above)
Comment (shashikantdurge) [reply android/build.gradle:37]: Hey @MaiKaY , I am not able to find any versions for this.
As per the segment docs :
 
Screenshot 2020-06-05 at 5 17 06 PM

**Comment by MaiKaY** [comment]
Okay then let's keep it as is. As soon as you implemented the configuration we are good to go to merge it.

**Comment by shashikantdurge** [comment]
@MaiKaY ENABLE_BRANCH_IO_INTEGRATION key added, Please review it

**Pull request merged**